templates/gitlab/mender-artifact-upload.gitlab-ci.yml

```diff
@@ -0,0 +1,24 @@
+# GitLab CI template to upload Mender artifacts from MENDER_ARTIFACTS_FOLDER folder
+# to a Mender server.
+#
+# The template is designed to be used with the include keyword or copied into an existing
+# configuration. It requires `MENDER_SERVER_URL` and `MENDER_SERVER_ACCESS_TOKEN` variables
+# to work and can be used to upload a Mender artifacts from `MENDER_ARTIFACTS_FOLDER` folder
+# to a Mender server.
+
+include:
+  - project: 'Northern.tech/Mender/mender-ci-workflows'
+    file: 'templates/gitlab/common.gitlab-ci.yml'
+    ref: 'master'
+
+variables:
+  MENDER_ARTIFACT_NAME: "" # Mender Artifact file name in the `MENDER_ARTIFACTS_FOLDER` folder
+
+.mender:upload:artifact:
+  image: mendersoftware/mender-ci-tools:mender-master
+  script:
```

Comment:
I would check for `MENDER_SERVER_URL` and `MENDER_SERVER_ACCESS_TOKEN` variables

It very common that `MENDER_SERVER_ACCESS_TOKEN` is a protected variable and hence not available in all pipelines, and if it is empty this template will fail the `mender-cli` command. But I think it is too late, it should check early that the variables are set.